Fix for visualising attached shapes

VladimirIvan opened the pull request:
Shapes created by defining custom links do not get published when calling publish_frames. This was because they were considered robot as robot geometry and got ignored. This avoids displaying the robot geometry twice, 1st as a part of the robot model and 2nd as the collision object published as a MarkerArray.
- Adds a separate logic to publish custom shapes attached to the robot but ignoring the shapes defined in the URDF.
- The links defined in the URDF are now correctly marked with `is_robot_link=true`. This is then used to distinguish links loaded from the URDF from custom links.

vlad-touchlab committed:
Fix for visualising attached shapes

VladimirIvan commented:
This wiil need rebasing after #756 is merged.

wxmerkt commented on exotica_core/src/scene.cpp:
```diff
@@ -783,6 +783,7 @@ void Scene::UpdateSceneFrames()
 
             std::string collision_element_name = links[i]->getName() + "_collision_" + std::to_string(j);
             std::shared_ptr<KinematicElement> element = kinematica_.AddElement(collision_element_name, trans, links[i]->getName(), links[i]->getShapes()[j]);
+            element->is_robot_link = true;
```
Is this necessary? In the collision scene, we always check the `closest_robot_link.lock()` on the `weak_ptr` to check if it is attached to a robot link?

wxmerkt replied:
Attached objects were always handled via the `closest_robot_link` instead of setting it to `is_robot_link` itself

vlad-touchlab committed:
Rebase fix for collision shapes on robot

VladimirIvan commented:
I checked the robot link checks. My edits were indeed not necessary.
I removed the changes and rebased the branch and this PR doesn't bring any useful changes. I'm closing this now
